### DISCUSSION TITLE
Final Stretch Meeting (4/9)

### DISCUSSION THREAD

**Discussion by Cdbryan** [Workspace-Recommendation-Engine/reco_eng#28] [General]
Hey guys! @thelearner411 and I met today and planned out a couple of major things left to complete as far as connecting the frontend to the backend 

1. We need to build out our database structure more to accommodate the model. I will attach a document with the proposed database structure 
2. For the cold start scenario we decided the user would be asked to rank workspace categories (cafe/bakery, library, and coworking space) these rankings would then be applied to workspaces that fit into that category. (Ex: User ranks library as 5, then workspaces that have a label of public library and university library would automatically have a rating of 5 when the user first uses the program) 
3. We then need to create a rating system that can be utilized once a user visits a workspace the rating from this would replace the rating that was automatically added in the coldstart scenario dependent on what type of workspace the user rated. (This requires a bit of work on both the frontend and backend) 
4. We also need to figure out how to sort items based on distance from the user on the matching page 
5. @thelearner411 is working on adding our workspaces to a json file so that I can pull from that file on the matches page 
6. We may need to manually add photos of all workspaces to the JSON file that holds the workspaces since we currently don't have any 

I know this is a lot of information so I think we should schedule a meeting on Monday or Tuesday to divide out tasks and complete this work as soon as possible! We are in the last stretch let's try to complete everything we hoped to! :) 
[Workspace Reco Database Design.pdf](https://github.com/Workspace-Recommendation-Engine/reco_eng/files/11190712/Workspace.Reco.Database.Design.pdf)

**Comment by farahabuatiyeh** [comment]
Thanks for the update and for working on proposed database structure document. It's great to see all the effort and work that you and @thelearner411 put and thanks for identifying the remaining tasks to complete.

Regarding the cold start scenario, the approach of asking users to rank workspace categories seems like a good way to generate initial ratings. We'll need to make sure that the backend and frontend are properly connected to execute this feature.

It's also important that we create a rating system that can be used by users after they visit a workspace. This will require work on both the backend and frontend, as you mentioned.

Sorting items based on distance from the user on the matching page is a critical feature that we need to get right. I appreciate @thelearner411's work on adding workspaces to the JSON file, and we can definitely explore adding photos of all workspaces to the file as well.

Let's meet on Tuesday at 17:00, as planned, to divide up the remaining tasks and ensure that we can complete everything we hoped to. 
Keep up the great work!

**Comment by Cdbryan** [comment]
Thanks for your comment @farahabuatiyeh! Just added the workspace database design to my original post let me know if you have any questions, see you on Tuesday!

**Reply by farahabuatiyeh** [reply]
The design looks great, thank you so much and see you soon!

**Comment by SoohyeonB** [comment]
@Cdbryan Modified erd for db! let me know if you guys have any questions or things to be changed!
![reco-engine](https://user-images.githubusercontent.com/56013677/231387457-dc0fa06c-c0aa-4c4b-a423-ea4d67380355.png)
If everyone thinks okay, then I will adjust to the firebase.

**Reply by Cdbryan** [reply]
Looks good to me thank you! @SoohyeonB

**Reply by farahabuatiyeh** [reply]
It's great, thank you!
